source/common/init/watcher_impl.cc

```diff
@@ -4,34 +4,38 @@ namespace Envoy {
 namespace Init {
 
 WatcherHandleImpl::WatcherHandleImpl(absl::string_view handle_name, absl::string_view name,
-                                     std::weak_ptr<ReadyFn> fn)
+                                     std::weak_ptr<TargetAwareReadyFn> fn)
     : handle_name_(handle_name), name_(name), fn_(std::move(fn)) {}
 
 bool WatcherHandleImpl::ready() const {
   auto locked_fn(fn_.lock());
   if (locked_fn) {
     // If we can "lock" a shared pointer to the watcher's callback function, call it.
     ENVOY_LOG(debug, "{} initialized, notifying {}", handle_name_, name_);
-    (*locked_fn)();
+    (*locked_fn)(handle_name_);
     return true;
   } else {
     // If not, the watcher was already destroyed.
-    ENVOY_LOG(debug, "{} initialized, but can't notify {} (unavailable)", handle_name_, name_);
+    ENVOY_LOG(debug, "{} initialized, but can't notify {}", handle_name_, name_);
     return false;
   }
 }
 
 WatcherImpl::WatcherImpl(absl::string_view name, ReadyFn fn)
-    : name_(name), fn_(std::make_shared<ReadyFn>(std::move(fn))) {}
+    : name_(name), fn_(std::make_shared<TargetAwareReadyFn>(
+                       [callback = std::move(fn)](absl::string_view) { callback(); })) {}
```

Comment:
Do we know what targets are relying on this path? Ideally I'd like to see all init targets are with names at the end. If there's a lot, can you file an issue to track?

Reply:
No, I don't think we can get the target names. Since the passed in callback is ```std::function<void()>```. And I suppose this type of watcher(Ctor with ```<void()>```) has not been given tasks to track the name.

Reply:
https://github.com/envoyproxy/envoy/pull/12035#discussion_r453977229 
The initial discussion to respect no target name as argument is above.

We can either choose passing target name everywhere, which is done at the initial commits,
or we allow function doesn't careabout target name as current

Reply:
My question is what is using this without name. They are basically tech debts, no?

Reply:
Watchers that are passed from higher level to a dynamic init manager is using this kind of Ctor. Callbacks of these watchers does not need to pass names. Commonly their callback is ```higher_level_target.ready()``` and ```ready()``` of these targets does not need names.

Reply:
So that's why I proposed to separate watchers into 2 types, one for targets and the other for init manager. Currently only those for init managers will need to pass names to managers.

Reply:
Yeah, I'm fine leaving those usages and keep them separately, just want to know what are using them.